attributes/default.rb

```diff
@@ -434,101 +434,6 @@
-                                             libffi-devel dkms mysql-devel libedit-devel postgresql-devel postgresql-server
-                                             sendmail cmake byacc libglvnd-devel mdadm libgcrypt-devel libevent-devel
-                                             libxml2-devel perl-devel tar gzip bison flex gcc gcc-c++ patch
-                                             rpm-build rpm-sign system-rpm-config cscope ctags diffstat doxygen elfutils
-                                             gcc-gfortran git indent intltool patchutils rcs subversion swig systemtap curl
-                                             jq wget python-pip NetworkManager-config-routing-rules libibverbs-utils
-                                             librdmacm-utils python3 python3-pip iptables libcurl-devel yum-plugin-versionlock
-                                             coreutils moreutils sssd sssd-tools sssd-ldap)
-
-    # Install R via amazon linux extras
-    default['cluster']['alinux_extras'] = ['R3.4']
-  end
-
-  default['cluster']['chrony']['service'] = "chronyd"
-  default['cluster']['chrony']['conf'] = "/etc/chrony.conf"
-
-when 'debian'
-  default['openssh']['server']['subsystem'] = 'sftp internal-sftp'
-  default['cluster']['base_packages'] = %w(vim ksh tcsh zsh libssl-dev ncurses-dev libpam-dev net-tools libhwloc-dev dkms
-                                           tcl-dev automake autoconf libtool librrd-dev libapr1-dev libconfuse-dev
-                                           apache2 libboost-dev libdb-dev tcsh libncurses5-dev libpam0g-dev libxt-dev
-                                           libmotif-dev libxmu-dev libxft-dev libhwloc-dev man-db lvm2 python
-                                           r-base libblas-dev libffi-dev libxml2-dev mdadm
-                                           libgcrypt20-dev libmysqlclient-dev libevent-dev iproute2 python3 python3-pip
-                                           libatlas-base-dev libglvnd-dev iptables libcurl4-openssl-dev
-                                           coreutils moreutils sssd sssd-tools sssd-ldap curl)
-
-  case node['platform_version']
-  when '18.04'
-    default['cluster']['base_packages'].push('python-pip', 'python-parted')
-  when '20.04'
-    default['cluster']['base_packages'].push('python3-parted')
-  end
-
-  # Modulefile Directory
-  default['cluster']['modulefile_dir'] = "/usr/share/modules/modulefiles"
-  # MODULESHOME
-  default['cluster']['moduleshome'] = "/usr/share/modules"
-  # Config file used to set default MODULEPATH list
-  default['cluster']['modulepath_config_file'] = "#{node['cluster']['moduleshome']}/init/.modulespath"
-  default['cluster']['kernel_headers_pkg'] = "linux-headers-#{node['kernel']['release']}"
-  default['cluster']['chrony']['service'] = "chrony"
-  default['cluster']['chrony']['conf'] = "/etc/chrony/chrony.conf"
-
-  if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
```

Comment:
We removed this if: what does it mean?

Reply:
It checks if the version of the platform is >=15.04 and it is in the debian branch: since we do not support ubuntu version less than 18.04 it is always true. I have check the if with an interactive ruby shell.